Data/models/Bookmark.swift

```diff
@@ -350,12 +350,15 @@ extension Bookmark {
             guard let bookmarkToUpdate = context.object(with: self.objectID) as? Bookmark else { return }
             
             // See if there has been any change
-            if bookmarkToUpdate.customTitle == customTitle && bookmarkToUpdate.url == url && bookmarkToUpdate.syncOrder == newSyncOrder, case .keep = location {
+            if bookmarkToUpdate.customTitle == customTitle &&
+                bookmarkToUpdate.url == url &&
+                (newSyncOrder == nil || bookmarkToUpdate.syncOrder == newSyncOrder),
```

Comment:
There was a bug that when no syncOrder was provided in update method, and the bookmark had a sync order set, it was passing through this condition